app/src/main/java/org/wikipedia/analytics/SuggestedEditsFunnel.java

```diff
@@ -2,35 +2,112 @@
 
 import android.support.annotation.NonNull;
 
+import com.google.gson.annotations.SerializedName;
+
 import org.json.JSONObject;
+import org.wikipedia.Constants;
 import org.wikipedia.WikipediaApp;
+import org.wikipedia.json.GsonUtil;
 
 public class SuggestedEditsFunnel extends TimedFunnel {
+    private static SuggestedEditsFunnel INSTANCE;
+
     private static final String SCHEMA_NAME = "MobileWikiAppSuggestedEdits";
-    private static final int REV_ID = 0; // TODO
+    private static final int REV_ID = 18949003;
 
     private static final String SUGGESTED_EDITS_UI_VERSION = "1.0";
     private static final String SUGGESTED_EDITS_API_VERSION = "1.0";
 
     public static final String SUGGESTED_EDITS_ADD_COMMENT = "#suggestededit-add " + SUGGESTED_EDITS_UI_VERSION;
     public static final String SUGGESTED_EDITS_TRANSLATE_COMMENT = "#suggestededit-translate " + SUGGESTED_EDITS_UI_VERSION;
 
-    public SuggestedEditsFunnel(WikipediaApp app) {
+    private Constants.InvokeSource invokeSource;
+    private String parentSessionToken;
+    private int helpOpenedCount = 0;
+    private int contributionsOpenedCount = 0;
+    private SuggestedEditStatsCollection statsCollection = new SuggestedEditStatsCollection();
+
+    public SuggestedEditsFunnel(WikipediaApp app, Constants.InvokeSource invokeSource) {
         super(app, SCHEMA_NAME, REV_ID, Funnel.SAMPLE_LOG_ALL);
+        this.invokeSource = invokeSource;
+        this.parentSessionToken = app.getSessionFunnel().getSessionToken();
+    }
+
+    public static SuggestedEditsFunnel get(Constants.InvokeSource invokeSource) {
+        if (INSTANCE == null) {
+            INSTANCE = new SuggestedEditsFunnel(WikipediaApp.getInstance(), invokeSource);
+        }
+        return INSTANCE;
+    }
+
+    public static SuggestedEditsFunnel get() {
+        return get(Constants.InvokeSource.NAV_MENU);
+    }
+
+    public static void reset() {
+        INSTANCE = null;
     }
 
     @Override protected void preprocessSessionToken(@NonNull JSONObject eventData) {
-        // TODO: preprocessData(eventData, "session_token", token);
+        preprocessData(eventData, "session_token", parentSessionToken);
+    }
+
+    public SuggestedEditStats getAddDescriptionStats() {
+        return statsCollection.addDescriptionStats;
+    }
+
+    public SuggestedEditStats getTranslateDescriptionStats() {
+        return statsCollection.translateDescriptionStats;
+    }
+
+    public void helpOpened() {
+        helpOpenedCount++;
     }
 
-    @Override
-    protected JSONObject preprocessData(@NonNull JSONObject eventData) {
-        preprocessData(eventData, "ui_version", SUGGESTED_EDITS_UI_VERSION);
-        preprocessData(eventData, "api_version", SUGGESTED_EDITS_API_VERSION);
-        return super.preprocessData(eventData);
+    public void contributionsOpened() {
+        contributionsOpenedCount++;
     }
```

Comment:
May be we can use this now that 'My Contributions' is ready..

Reply:
Done